pkg/kubectl/cmd/cmd.go

```diff
@@ -598,6 +601,57 @@ func NewKubectlCommand(in io.Reader, out, err io.Writer) *cobra.Command {
 		}
 	}
 
+	cmdutil.CheckErr(cmds.RegisterFlagCompletionFunc("namespace", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
```

Comment:
Any reason why this `RegisterFlagCompletionFunc` is done differently than the other three below?

Reply:
because we could not get namespaces from local kubeconfig.